scripts/superdesk/upload/image-crop-directive.js

```diff
@@ -279,136 +278,107 @@
-                        left: (scope.point.x * img.width) + 19
+                    element.find('.image-point__poi__cross-right').css({
+                        width: img.width - (leftOffset + (2 * circleRadius)),
+                        top: horizontalTopffset,
+                        left: leftOffset +  (2 * circleRadius)
                     });
-                    crossBottom.css({
-                        height: ((1 - scope.point.y) * img.height) - 19,
-                        top: (scope.point.y * img.height) + 19,
-                        left: (scope.point.x * img.width) + 1
+                    element.find('.image-point__poi__cross-top').css({
+                        height: topOffset,
+                        left: verticalLeftOffset
                     });
-                    crossTop.css({
-                        height: (scope.point.y * img.height) - 15,
-                        left: (scope.point.x * img.width) + 1
+                    element.find('.image-point__poi__cross-bottom').css({
+                        height: img.height - (topOffset + (2 * circleRadius)),
+                        left: verticalLeftOffset,
+                        top: topOffset + (2 * circleRadius)
                     });
                 }
+                // init directive element style
+                element.css({
+                    position: 'relative',
+                    display: 'block'
+                });
+                // load the image in order to know the size
+                var img = new Image();
+                img.onload = function() {
+                    var $img = element.find('.image-point__image').get(0);
+                    drawPoint($img);
+                    // draw when needed
+                    scope.$on('poiUpdate', function() {
+                        drawPoint($img);
+                    });
+                    angular.element($window).on('resize', function() {
+                        drawPoint($img);
+                    });
+                    // setup overlay to listen mouse events
+                    (function($img) {
+                        function updatePOI(e) {
+                            vm.updatePOI({
+                                x: Math.round(e.offsetX * 100 / $img.width) / 100,
+                                y: Math.round(e.offsetY * 100 / $img.height) / 100
+                            });
+                        }
+                        var overlay = element.find('.image-point__poi__overlay');
+                        var mousedown = false;
+                        overlay.on('mousedown', function(e) {
+                            // enable drag mode
+                            mousedown = true;
+                            updatePOI(e);
+                        });
+                        overlay.on('mousemove', function(e) {
+                            if (mousedown) {
+                                updatePOI(e);
```

Comment:
Is it needed to update on mousemove? It seems a bit more efficient to update on mouseleave/mouseup. I'm not sure if the final mousemove will totally match the mouseup event.